### DISCUSSION TITLE
facet_ranges in facets

### DISCUSSION THREAD

**Discussion by FrancescoBrunoDev** [vufind-org/vufind#3423] [Q&A]
Hi everyone,

Is it possible to configure the facets.ini file to display facet_ranges as well? For example, I want to have a facet that allows selecting records based on their last modification date. I have already set up everything on the Solr side and, with the query:
```
select?q=*:*&facet=true&facet.range=modifydate&rows=0&facet=true&facet.range.start=NOW/DAY-6MONTHS&facet.range.end=NOW&facet.range.gap=%2B3MONTH&facet.limit=30&facet.sort=count&facet.mincount=1&wt=json
```
I receive the following response:
```
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_ranges":{
      "modifydate":{
        "counts":[
          "2023-08-16T00:00:00Z",102824,
          "2023-11-16T00:00:00Z",32037],
        "gap":"+3MONTH",
        "start":"2023-08-16T00:00:00Z",
        "end":"2024-05-16T00:00:00Z"}},
    "facet_intervals":{},
    "facet_heatmaps":{}}}
```
Analyzing the codebase, I don't seem to find any reference to facet_ranges that makes me optimistic. 
Thank you for any help!

**Comment by demiankatz** [comment]
Unfortunately, I believe you're correct that this is not currently supported. It should certainly be possible to add -- it would just require some processing logic in VuFind\Search\Solr\Params and associated facet display logic. Probably not a trivial process, though... and it's also complicated a little bit by the fact that we need to revise all of our facet logic due to Solr 9 deprecations (see [VUFIND-1210](https://openlibraryfoundation.atlassian.net/browse/VUFIND-1210)). Might be easier to reduce technical debt before adding new features, but of course if this is an urgent need, I'm not opposed to working on this sooner. Feel free to open a JIRA ticket or pull request, depending on how much help you need to implement this. :-)
